code/form/WidgetAreaEditor.php

```diff
@@ -50,15 +50,17 @@ public function AvailableWidgets()
                 unset($classes[0]);
             }
             
-            foreach ($classes as $class) {
-                $available = Config::inst()->get($class, 'only_available_in');
-                
-                if (!empty($available) && is_array($available)) {
-                    if (in_array($this->Name, $available)) {
+            if( isset($classes) ){
```

Comment:
Seems like `null` is a valid return type of `ClassInfo::subclassesFor` in SS3. I'd prefer the fix to be 
```
if (!is_array($classes)) {
    continue;
}
```
right after the line that calls `ClassInfo::subclassesFor`.

Reply:
I also think that casting `$classes` as an array would be a one line fix. This could be done on line 45, or in the foreach statement. Example https://eval.in/1055120